common-utils/src/common/deployment/__init__.py

```diff
@@ -497,27 +668,55 @@ def process_project_flows(self, validate_only: bool = False) -> None:
         Raises:
             Exception: Exception will thrown for bad FLOW_SPEC definitions.
         """
+        # account_id and ecs client is only populated on deployment
+        # need default values for validation to work and prevent unbound variables
+        account_id = ""
+        ecs_client = object()
         flows_path = self._prefect_flows_folder
+        # setup to aggregate errors for raising at the end as needed
         flow_errors = []
         missing_flow_specs = []
+        # this is the deployment path
         if not validate_only:
+            # command type used for DRY logging
             command_type = "deployment"
+            # only need AWS for deployment
+            # single account_id and ecs client for entire deployment workflow
+            from boto3 import session
+
+            init_session = session.Session()
+            account_id = get_aws_account_id(init_session)
+            ecs_client = init_session.client("ecs")
```

Comment:
only doing the AWS work once and only on deployment.  This means that local flow development and `--build_only` and `--validate_only` flags on the `deploy-cli` do not need AWS access or libraries.